src/reyna/Blackout/BlackoutTime.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Reyna
+{
+    using System;
+    using System.Globalization;
+
+    public class BlackoutTime
+    {
+        public bool CanSendAtTime(DateTime now, string ranges)
+        {
+            if (String.IsNullOrEmpty(ranges))
+            {
+                return true;
+            }
+
+            string[] rangesSplit = ranges.Split(',');
+            foreach (string range in rangesSplit)
+            {
+                TimeRange timeRange = this.ParseTime(range);
+                Time timeNow = new Time(now.Hour, now.Minute);
+                if (timeRange.Contains(timeNow))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public TimeRange ParseTime(string range)
+        {
+            string[] rangeSplit = range.Split('-');
+            string format = "HH:mm";
+
+            DateTime from = DateTime.ParseExact(rangeSplit[0], format, CultureInfo.InvariantCulture);
+            DateTime to = DateTime.ParseExact(rangeSplit[1], format, CultureInfo.InvariantCulture);
```

Comment:
This case is covered by the IsBlackoutRangeValid function in preferences, as we validate every time we cave a range to the preferences.

I have added a test case to make sure in this commit: https://github.com/B2MSolutions/reyna.net/commit/5a62df01d2edef7c06cba2e93c47b9ece1ad38cc
@CharlesGarth